src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx

```diff
@@ -91,34 +93,36 @@ export const ProjectFeatureToggles = ({
             }) as ListItemType[];
         }
 
-        return features.map(
-            ({
-                name,
-                lastSeenAt,
-                createdAt,
-                type,
-                stale,
-                environments: featureEnvironments,
-            }) => ({
-                name,
-                lastSeenAt,
-                createdAt,
-                type,
-                stale,
-                environments: Object.fromEntries(
-                    environments.map(env => [
-                        env,
-                        {
-                            name: env,
-                            enabled:
-                                featureEnvironments?.find(
-                                    feature => feature?.name === env
-                                )?.enabled || false,
-                        },
-                    ])
-                ),
-            })
-        );
+        return features
+            .slice(0, features.length > limit ? pageSize : limit)
```

Comment:
For projects above 300-500 items, there are issues rendering this page. I didn't virtualize it yet, so this is a stop-gap solution.

Reply:
This will impact our demo instance. We have over 300 feature toggles there, might not be the best experience for new users, can we paginate instead in the interim?